pints/toy/_toy_classes.py

```diff
@@ -46,6 +48,22 @@ class ToyModel(object):
     model classes, e.g. :class:`pints.ForwardModel`.
     """
 
+    def _dfdp(self, y, t, p):
+        """
+        Returns the derivative of the ODE RHS with respect to parameters.
+        This should be a matrix of dimensions ``n_parameters`` by
+        ``n_parameters``.
+        """
+        raise NotImplementedError
+
+    def jacobian(self, y, t, p):
```

Comment:
What's the rationale for having `jacobian` public and the others private?

Reply:
Jacobians are useful for other things (like some gradient-based optimisers and samplers), so thought I'd give them a public method.